### PR TITLE
add newtype for TaskArgs, TaskModule, TaskName, TaskModuleContents, D…

### PR DESCRIPTION
**Pull request by colinwahl**
…river, CommandSumType, ImportList

This is my first crack at adding newtypes to replace our strings.

Let me know where the best placement for the newtypes is, I never know exactly where they should go. Maybe the top of the file?

Also let me know if I should change the naming of the field accessors.

### CODE REVIEW

**Comment by parsonsmatt** [comment]
@colinwahl can you merge most recent master to get the travis build going?

**Comment by colinwahl** [comment]
@parsonsmatt Just to clarify - are the tests failing because of hlint?

**Comment by parsonsmatt** [comment]
Yes, they are. You can run `hlint .` to check the hints locally.

File: src/Kale.hs [hunk 1]
```diff
@@ -137,9 +151,9 @@ fileToTask dir file = runMaybeT $
 
 mkTask :: String -> String -> String -> Task
 mkTask fileContent name mod_ = Task
-    { taskModule = mod_
-    , taskArgs   = findArgs fileContent
-    , taskName   = casify name
+    { taskModule = TaskModule mod_
+    , taskArgs   = TaskArgs <$> findArgs fileContent
+    , taskName   = TaskName $ casify name
```
Comment (parsonsmatt) [inline src/Kale.hs]: I would prefer to see the `mkTask` function take the more strongly typed stuff -- eg `mkTask :: FileContent -> TaskName -> TaskModule -> Task`. Of course, then it's just `Task` :)

We could expose a function `mkTaskName = TaskName . casify` and have that be the only exported function. That way we know that any string in a `TaskName` is already cased appropriately.

Likewise, `mkTaskArgs :: FileContent -> Maybe TaskArgs` could be used as the smart constructor for the task arguments. In fact, we may want to make `TaskArgs` a sum type of `NoArguments | RecordArgs String | PositionalArgs String` :)

**Review by parsonsmatt — CHANGES_REQUESTED** [review]
There's one change I'd like to see (pushing the newtype wrapping further to the edge of the program). Otherwise, this is great! Thanks for the PR :smile:

File: src/Kale.hs (hunk 1 above)
Comment (colinwahl) [reply src/Kale.hs]: Just added the changes you requested, let me know what you think.

Comment (colinwahl) [reply src/Kale.hs]: I wasn't sure how to detect if there were record vs positional args.  I took a guess at how to do it, but I don't think it is the best solution.

File: test/KaleSpec.hs [hunk 2]
```diff
@@ -56,12 +56,12 @@ spec = do
...
         it "is empty on empty task lists" $
-            mkCommandSum [] `shouldBe` ""
+            unCommandSumType (mkCommandSum []) `shouldBe` ""
         it "lists tasktnames properly" $
-            mkCommandSum [task0, task1]
+            unCommandSumType (mkCommandSum [task0, task1])
                 `shouldBe` concat
                     [ "data Command = Name { foo :: Int }"
-                    , " | Other_Name deriving (Eq, "
+                    , " | OtherName deriving (Eq, "
```
Comment (parsonsmatt) [inline test/KaleSpec.hs]: I want the underscores in the name. That way the commands translate to `other-name` instead of `othername`.

**Review by parsonsmatt — CHANGES_REQUESTED** [review]
Awesome! Just one more change -- the task command arg should have underscores in it.

We are not correctly handling positional arguments at all. That can come in a later patch.

Thanks!

File: test/KaleSpec.hs (hunk 2 above)
Comment (colinwahl) [reply test/KaleSpec.hs]: Ah - I see. Fixing now

**Comment by colinwahl** [comment]
@parsonsmatt This job https://travis-ci.org/parsonsmatt/kale/jobs/305572373 seems to have failed from a TimeoutError.  Is there a way to kick it off again?

**Review by parsonsmatt — APPROVED** [review]
Woderful, thanks! :smile:

**Pull request merged**